File: 240622/특정 문자로 시작하는 문자열/strings-that-start-with-a-specific-character.py
N = int(input())
words = [input() for _ in range(N)]
find_chr = input()

cnts, avg_len = 0, 0
answer = []
for word in words:
    if word.startswith(find_chr):
        answer.append(word)
        cnts += 1

avg_len = sum([len(word) for word in answer]) / len(answer)

print(cnts, "{:.2f}".format(avg_len))

File: 251130/특정 문자로 시작하는 문자열/strings-that-start-with-a-specific-character.cpp
#include <iostream>
#include <vector>
#include <string>
#include <iomanip>
using namespace std;

int main() 
{
    int N;
    cin >> N;

    vector<string> words(N);
    for (int i = 0; i < N; i++) 
    {
        cin >> words[i];
    }

    string find_chr;
    cin >> find_chr;

    int cnts = 0;
    int total_len = 0;

    for (const string& word : words) 
    {
        if (word.size() >= find_chr.size() && word.substr(0, find_chr.size()) == find_chr) 
        {
            cnts++;
            total_len += word.length();
        }
    }

    double avg_len = (cnts == 0) ? 0.0 : (double)total_len / cnts;

    cout << cnts << " " << fixed << setprecision(2) << avg_len;
    return 0;
}
